core/encoding/uuid/definitions.odin

```diff
@@ -0,0 +1,59 @@
+package uuid
+
+// A RFC 4122 Universally Unique Identifier
+Identifier :: struct #raw_union {
+	integer: u128be,
+	bytes:   [16]u8,
+}
```

Comment:
I am not sure if it's better to have it be a raw union like this or just:
```odin
Identifier :: distinct [16]u8
```

Reply:
The only thing `.integer` appears to be used for is for the pseudo-random variant's generation, to use `rand_u128`. I think that `Identifier` should just be `distinct [16]u8`. `generate_v4` can transmute the u128 to `[16]u8`.

Also, because it's pseudorandom, enforcing that you're putting the random bits in in a big endian order instead of machine endian is a bit overkill, so `u128` -> `[16]u8` is just fine.

Reply:
And even then, `rand.read` is a much better idea too.
```
_ = rand.read(identifier[:])
```

Reply:
I can address both concerns.